File: VB/Rebill/Get_Recurring_Payment_Status.vb
' *
' * Bluepay VB.NET Sample code.
' *
' * This code sample runs a $0.00 Credit Card Auth transaction
' * against a customer using test payment information.
' * Once the rebilling cycle is created, this sample shows how to
' * Get information back on this rebilling cycle.
' * See comments below on the details of the initial Setup of the
' * rebilling cycle.
' *

Imports System
Imports BluePay.BPVB

Namespace BP10Emu

    Public Class Get_Recurring_Payment_Status

        Public Sub New()
            MyBase.New()
        End Sub

        Public Shared Sub Main()

            Dim accountID As String = "100221257489"
            Dim secretKey As String = "YCBJNEUEKNINP5PWEH1HRQDSQHYANPM/"
            Dim mode As String = "TEST"

            ' Merchant's Account ID
            ' Merchant's Secret Key
            ' Transaction Mode: TEST (can also be LIVE)
            Dim payment As BluePayPayment_BP10Emu = New BluePayPayment_BP10Emu(
                    accountID,
                    secretKey,
                    mode)

            ' Card Number: 4111111111111111
            ' Card Expire: 12/15
            ' Card CVV2: 123
            payment.SetCCInformation(
                    "4111111111111111",
                    "1215",
                    "123")

            ' First Name: Bob
            ' Last Name: Tester
            ' Address1: 123 Test St.
            ' Address2: Apt #500
            ' City: Testville
            ' State: IL
            ' Zip: 54321
            ' Country: USA
            payment.SetCustomerInformation(
                    "Bob",
                    "Tester",
                    "123 Test St.",
                    "Apt #500",
                    "Testville",
                    "IL",
                    "54321",
                    "USA")

            ' Rebill Amount: $3.50
            ' Rebill Start Date: Jan. 5, 2015
            ' Rebill Frequency: 1 MONTH
            ' Rebill # of Cycles: 5
            payment.SetRebillingInformation(
                    "3.50",
                    "2015-01-05",
                    "1 MONTH",
                    "5")

            ' Phone #: 123-123-1234
            payment.SetPhone("123-123-1234")

            ' Email Address: test@bluepay.com
            payment.SetEmail("test@bluepay.com")

            ' Auth Amount: $0.00
            payment.Auth("0.00")

            Dim result As String = payment.Process()

            ' If transaction was approved..
            If (result = "1") Then

                Dim rebillGetStatus As BluePayPayment_BP10Emu = New BluePayPayment_BP10Emu(
                        accountID,
                        secretKey,
                        mode)

                ' Cancels rebill using Rebill ID token returned
                rebillGetStatus.GetRebillStatus(payment.GetRebillID())

                rebillGetStatus.Process()

                ' Outputs response from BluePay gateway
                Console.Write("Rebill ID: " + rebillGetStatus.GetRebillID() + Environment.NewLine)
                Console.Write("Rebill Status: " + rebillGetStatus.GetStatus() + Environment.NewLine)
                Console.Write("Rebill Creation Date: " + rebillGetStatus.GetCreationDate() + Environment.NewLine)
                Console.Write("Rebill Next Date: " + rebillGetStatus.GetNextDate() + Environment.NewLine)
                Console.Write("Rebill Last Date: " + rebillGetStatus.GetLastDate() + Environment.NewLine)
                Console.Write("Rebill Schedule Expression: " + rebillGetStatus.GetSchedExpr() + Environment.NewLine)
                Console.Write("Rebill Cycles Remaining: " + rebillGetStatus.GetCyclesRemain() + Environment.NewLine)
                Console.Write("Rebill Amount: " + rebillGetStatus.GetRebillAmount() + Environment.NewLine)
                Console.Write("Rebill Next Amount: " + rebillGetStatus.GetNextAmount() + Environment.NewLine)
            Else
                Console.Write(payment.GetMessage())
            End If
        End Sub
    End Class
End Namespace

File: C++/Rebill/Get_Recurring_Payment_Status.cpp
//
// BluePay C++ Sample code.
//
// This code sample runs a $0.00 Credit Card Auth transaction
// against a customer using test payment information.
// Once the rebilling cycle is created, this sample shows how to
// get information back on this rebilling cycle.
// See comments below on the details of the initial setup of the
// rebilling cycle.

#include "Get_Recurring_Payment_Status.h"
#include "../bluepay-cpp/BluePay.h"
#include <iostream>
using namespace std;

void getRecurringPaymentStatus(){
    string accountId = "Merchant's Account ID Here";
    string secretKey = "Merchant's Secret Key Here";
    string mode = "TEST";
	
    BluePay rebill(
    	accountId, 
	    secretKey, 
	    mode 
    );

    rebill.setCustomerInformation(
	    "Bob", // First Name
	    "Tester", // Last Name
	    "1234 Test St.", // Address1
	    "Apt #500", // Address2
	    "Testville", // City
	    "IL", // State
	    "54321", // Zip
	    "USA", // Country
	    "1231231234", // Phone Number
	    "test@bluepay.com" // Email Address
	  );

    rebill.setCCInformation(
	    "4111111111111111", // Card Number
	    "1225", // Card Expire
	    "123" // Card CVV2
    );

	rebill.setRebillingInformation(
		"15.00", // Rebill Amount: $3.50
		"2015-01-01", // Rebill Start Date: Jan. 5, 2015
		"1 MONTH", // Rebill Frequency: 1 MONTH
		"12" // Rebill # of Cycles: 5
	);

	// Auth Amount: $0.00
	rebill.auth("0.00");
    
    // Makes the API Request with Blue
	rebill.process();

	// If transaction was approved..
	if (rebill.isSuccessfulTransaction()){
		
		BluePay rebillStatus(
			accountId,
			secretKey,
			mode
		);

		// Find the rebill by ID and get rebill status 
		rebillStatus.getRebillStatus(rebill.getRebillId());

		// Makes the API Request to get the rebill status
		rebillStatus.process();

		// Reads the responses from BluePay if transaction was approved
		cout << string("Rebill Status: ") + rebillStatus.getStatus() + "\n";
		cout << string("Rebill ID: ") + rebillStatus.getRebillId() + "\n";
		cout << string("Rebill Creation Date: ") + rebillStatus.getCreationDate() + "\n";
		cout << string("Rebill Next Date: ") + rebillStatus.getNextDate() + "\n";
		cout << string("Rebill Last Date: ") + rebillStatus.getLastDate() + "\n";
		cout << string("Rebill Schedule Expression: ") + rebillStatus.getSchedExpr() + "\n";
		cout << string("Rebill Cycles Remaining: ") + rebillStatus.getCyclesRemain() + "\n";
		cout << string("Rebill Amount: ") + rebillStatus.getRebillAmount() + "\n";
	}
	else {
	    cout << string("Error: ") + rebill.getMessage();
	}

}